src/pages/iou/MoneyRequestSelectorPage.js

```diff
@@ -0,0 +1,147 @@
+import {withOnyx} from 'react-native-onyx';
+import {View} from 'react-native';
+import React, {useRef} from 'react';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import {createMaterialTopTabNavigator} from '@react-navigation/material-top-tabs';
+import ONYXKEYS from '../../ONYXKEYS';
+import FullPageNotFoundView from '../../components/BlockingViews/FullPageNotFoundView';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import TabSelector from '../../components/TabSelector/TabSelector';
+import CONST from '../../CONST';
+import useLocalize from '../../hooks/useLocalize';
+import * as IOUUtils from '../../libs/IOUUtils';
+import Navigation from '../../libs/Navigation/Navigation';
+import styles from '../../styles/styles';
+import MoneyRequestAmount from './steps/MoneyRequestAmount';
+import ReceiptSelector from './ReceiptSelector';
+import * as IOU from '../../libs/actions/IOU';
+import DragAndDropProvider from '../../components/DragAndDrop/Provider';
+import OnyxTabNavigator from '../../libs/Navigation/OnyxTabNavigator';
+import Permissions from '../../libs/Permissions';
+
+const TopTab = createMaterialTopTabNavigator();
+
+const propTypes = {
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            iouType: PropTypes.string,
+            reportID: PropTypes.string,
+        }),
+    }),
+
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    iou: PropTypes.shape({
+        id: PropTypes.string,
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+
+    /** Which tab has been selected */
+    selectedTab: PropTypes.string,
+};
+
+const defaultProps = {
+    betas: [],
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    selectedTab: CONST.TAB.MANUAL,
+};
+
+function MoneyRequestSelectorPage(props) {
+    const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
```

Comment:
This feels like a code smell to me. Why do we need `useRef` here?

Reply:
Honestly no clue, I just copied this from the other MoneyRequest pages.

Reply:
Hmmm yeah I don't think this is quite right. I think we should just remove the `useRef` from `iouType`, `reportID`, and `isEditing`, update their references throughout the rest of the file to remove `.current`, and then add them to dependency arrays as needed.

Reply:
Removed the useRef 👍